src/units/abilities.cpp

```diff
@@ -1169,17 +1169,23 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 }
 
 //Functions used for change damage_type list with damage
-static std::vector<std::string> damage_type_list(const unit_ability_list& abil_list, const std::string& type)
+static std::vector<std::string> damage_type_list(const unit_ability_list& abil_list, const std::string& type, unit_const_ptr& u, bool is_alternative)
```

Comment:
The caller checks empty() and then only uses front(). Is there a reason to return a std::vector instead of a std::optional?